### PR TITLE
[CYW30739] Support OTA requestor app.

### PR DESCRIPTION
**Pull request by hsusid**
#### Problem
Support CYW30739 OTA requestor app.

#### Change overview
* Add examples/ota-requestor-app/cyw30739 files.
* Add OTAImageProcessorImpl for CYW30739.
* Refine README files for CYW30739 apps.

#### Testing
```
chip-tool pairing ble-thread 1234 ...
chip-tool pairing onnetwork-vendor 4321 20202021 9050
chip-tool otasoftwareupdaterequestor announce-ota-provider 4321 9 0 0 1234 0
```

### CODE REVIEW

**Review by Damian-Nordic — APPROVED** [review]
Do you really need a separate OTA requestor app? IMHO every Matter-compliant device can be the OTA requestor and several platforms, including ours, successfully use lighting-app as the OTA requestor, so I think it's worth rethinking if adding and maintaining yet another example is worthy :)

**Comment by hsusid** [comment]
rebase

**Comment by hsusid** [comment]
> Do you really need a separate OTA requestor app? IMHO every Matter-compliant device can be the OTA requestor and several platforms, including ours, successfully use lighting-app as the OTA requestor, so I think it's worth rethinking if adding and maintaining yet another example is worthy :)

Thank you for the review. We are working on integrating OTA with CYW30739 lighting-app. Before it's got ready, the OTA requestor app is really helpful for us to do some OTA-related functionality verifications for cyw30739_sdk.

File: src/platform/CYW30739/OTAImageProcessorImpl.cpp [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+
+CHIP_ERROR OTAImageProcessorImpl::SetBlock(ByteSpan & block)
+{
+    if ((block.data() == nullptr) || block.empty())
+    {
+        return CHIP_NO_ERROR;
+    }
+
+    // Allocate memory for block data if it has not been done yet
+    if (mBlock.empty())
```
Comment (bzbarsky-apple) [inline src/platform/CYW30739/OTAImageProcessorImpl.cpp:182]: This implementation of SetBlock is buggy.  Please copy one of the non-buggy ones (e.g. Linux).

**Review by bzbarsky-apple — DISMISSED** [review]

File: src/platform/CYW30739/OTAImageProcessorImpl.cpp (hunk 1 above)
Comment (hsusid) [reply src/platform/CYW30739/OTAImageProcessorImpl.cpp]: Fixed.

**Review by selissia — APPROVED** [review]

File: src/platform/CYW30739/OTAImageProcessorImpl.cpp [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+        ChipLogError(SoftwareUpdate, "ImageProcessor context is null");
+        return;
+    }
+    else if (imageProcessor->mDownloader == nullptr)
+    {
+        ChipLogError(SoftwareUpdate, "mDownloader is null");
+        return;
+    }
+
+    const uint32_t written = wiced_firmware_upgrade_process_block(imageProcessor->mParams.downloadedBytes,
```
Comment (holbrookt) [inline src/platform/CYW30739/OTAImageProcessorImpl.cpp:161]: need a null check for `mBlock`? Is it possible for it to be null here?

Comment (hsusid) [reply src/platform/CYW30739/OTAImageProcessorImpl.cpp]: Fixed.

**Comment by hsusid** [comment]
rebase

**Review by andy31415 — APPROVED** [review]

**Pull request merged**